wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditManager.java

```diff
@@ -0,0 +1,196 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.api.audit.server;
+
+import com.codenvy.api.license.CodenvyLicense;
+import com.codenvy.api.license.LicenseException;
+import com.codenvy.api.license.server.CodenvyLicenseManager;
+import com.codenvy.api.permission.server.PermissionManager;
+import com.codenvy.api.permission.server.PermissionsImpl;
+import com.codenvy.api.user.server.dao.AdminUserDao;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.Files;
+
+import org.apache.commons.io.FileUtils;
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.Page;
+import org.eclipse.che.api.core.Page.PageRef;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.user.server.model.impl.UserImpl;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Facade for audit report related operations.
+ *
+ * @author Igor Vinokur
+ */
+@Singleton
+public class AuditManager {
```

Comment:
A general idea: separate a report printing logic and a retrieving business data.